### PR TITLE
Consolidate per function/data type integration test

### PR DESCRIPTION
**Pull request by xiangfu0**
Per: https://github.com/apache/pinot/issues/11298

Consolidate those tests requires custom schema/table creation/ data/segment builds into the same test suite to reduce the friction of bring up/down pinot components.

Each test can still run individually, and multiple tests will be running in the same test suite.

### CODE REVIEW

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      throws IOException {
+    LOGGER.info("Tearing down integration test class: {}", getClass().getSimpleName());
+    dropOfflineTable(getTableName());
+    FileUtils.deleteDirectory(_tempDir);
+    LOGGER.info("Finished tearing down integration test class: {}", getClass().getSimpleName());
+  }
+
+  public String getZkUrl() {
+    return ZkStarter.getDefaultZkStr();
+  }
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java:105]: Do we need to override `getZkUrl()` and `getBrokerBaseApiUrl()`?

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    return "http://localhost:" + DEFAULT_BROKER_PORT;
+  }
+
+  public TableConfig getTableConfig() {
+    return new TableConfigBuilder(TableType.OFFLINE).setTableName(getTableName()).build();
+  }
+
+  public abstract String getTableName();
+
+  public abstract Schema getSchema();
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java:121]: This should be `createSchema()`

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+  public String getHelixClusterName() {
+    return "CustomDataQueryClusterIntegrationTest";
+  }
+
+  protected String getBrokerBaseApiUrl() {
+    return "http://localhost:" + DEFAULT_BROKER_PORT;
+  }
+
+  public TableConfig getTableConfig() {
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java:115]: We should be able to use the existing `createOfflineTableConfig()`

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/CustomDataQueryClusterIntegrationTest.java:105]: yes, cause the zk/controller/broker/servers are created in the first test class, for other tests, those controller/broker/server are not created.

**Review by Jackie-Jiang — COMMENTED** [review]
Seems the suite is setup and tear down for each individual test. Also, these tests are actually split into 2 groups when running the CI

**Comment by xiangfu0** [comment]
> Seems the suite is setup and tear down for each individual test. Also, these tests are actually split into 2 groups when running the CI

Separated integration tests to different profiles then run them separately.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**